Stardew Valley - Health Checks fail when manifests use UTF-8-BOM

# Bug Report

## Summary
Reported by [@BucketListMod](https://discordapp.com/users/198556278790815744) on Discord.

When installing certain mods the Health Check that checks dependencies will throw a false positive and continue to tell the user they need to install a mod that is already in their loadout. This appears to be caused by mod authors using UTF-8-BOM instead of UTF-8 in their manifest.json 

According to the logs, the app was tripped up by the first character of the manifest.json file so it tracks that this is the BOM characters. We cater for this in [Vortex](https://github.com/Nexus-Mods/vortex-games/blob/8538000cf6baf839b7edd810345faf599d0acb94/game-stardewvalley/util.ts#L9). 

## Steps to reproduce
1. Install SMAPI.
2. Install [Diverse Seasonal Outfits (DSV)](https://www.nexusmods.com/Core/Libs/Common/Widgets/DownloadPopUp?id=109650&game_id=1303)
3. See the Health Check regarding "Content Patcher" being missing.
4. Install [ContentPatcher](https://www.nexusmods.com/Core/Libs/Common/Widgets/DownloadPopUp?id=105346&nmm=1&game_id=1303)
5. Note that the warnings for Content Patcher do not go away.
6. Check the app log files for the error

## What is the expected behaviour?
Manifests can be read if they are using UTF-8-BOM


## Other information
![image](https://github.com/user-attachments/assets/f116913a-283e-4a7d-a571-521b96b96632)

[Change that introduced this issue](https://github.com/Nexus-Mods/NexusMods.App/pull/2116/files#diff-9c5471db8bf33a6b8040d6610689f9f0995d89f4e00cf0051cc08d5de7b5a679R40)


[Content Patcher manifest.json (UTF-8-BOM)](https://github.com/user-attachments/files/17414017/manifest.json)

[Content Patcher manifest.json (UTF-8)](https://github.com/user-attachments/files/17414021/manifest.json)


[Solution on StackOverflow](https://stackoverflow.com/a/1319226)

```
00:23:42.069 [ERROR] (NexusMods.Games.StardewValley.Emitters.VersionDiagnosticEmitter) Exception while getting manifest for `Content Patcher`|Newtonsoft.Json.JsonReaderException: Unexpected character encountered while parsing value: ﻿. Path '', line 0, position 0.
   at Newtonsoft.Json.JsonTextReader.ParseValue()
   at Newtonsoft.Json.JsonReader.ReadForType(JsonContract contract, Boolean hasConverter)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Deserialize(JsonReader reader, Type objectType, Boolean checkAdditionalContent)
   at Newtonsoft.Json.JsonSerializer.DeserializeInternal(JsonReader reader, Type objectType)
   at Newtonsoft.Json.JsonConvert.DeserializeObject(String value, Type type, JsonSerializerSettings settings)
   at Newtonsoft.Json.JsonConvert.DeserializeObject[T](String value, JsonSerializerSettings settings)
   at StardewModdingAPI.Toolkit.Serialization.JsonHelper.Deserialize[TModel](String json) in /_/extern/SMAPI/src/SMAPI.Toolkit/Serialization/JsonHelper.cs:line 121
   at NexusMods.Games.StardewValley.Pipelines.<>c.<CreateManifestPipeline>b__3_0(Unit _, Hash _, Resource`1 resource, CancellationToken _) in /_/src/Games/NexusMods.Games.StardewValley/Pipelines.cs:line 42
   at NexusMods.Abstractions.Resources.Extensions.<>c__3`4.<<ThenDo>b__3_0>d.MoveNext() in /_/src/Abstractions/NexusMods.Abstractions.Resources/Extensions.cs:line 63
--- End of stack trace from previous location ---
   at NexusMods.Abstractions.Resources.Caching.ResourceCache`3.AddAsync(ResourceCache`3 self, TResourceIdentifier resourceIdentifier, CancellationToken cancellationToken)
   at BitFaster.Caching.Lru.ConcurrentLruCore`5.GetOrAddAsync[TArg](K key, Func`3 valueFactory, TArg factoryArgument)
   at NexusMods.Abstractions.Resources.Caching.ResourceCache`3.LoadResourceAsync(TResourceIdentifier resourceIdentifier, CancellationToken cancellationToken) in /_/src/Abstractions/NexusMods.Abstractions.Resources.Caching/ResourceCache.cs:line 40
   at NexusMods.Games.StardewValley.Emitters.Helpers.GetAllManifestsAsync(ILogger logger, ReadOnly loadout, IResourceLoader`2 pipeline, Boolean onlyEnabledMods, CancellationToken cancellationToken)+MoveNext() in /_/src/Games/NexusMods.Games.StardewValley/Emitters/Helpers.cs:line 52
```

[nexusmods.app.main.current.log](https://github.com/user-attachments/files/17414004/nexusmods.app.main.current.log)
[nexusmods.app.slim.current.log](https://github.com/user-attachments/files/17414006/nexusmods.app.slim.current.log)

--- a/src/Games/NexusMods.Games.StardewValley/Pipelines.cs
+++ b/src/Games/NexusMods.Games.StardewValley/Pipelines.cs
@@ -33,11 +33,11 @@ internal static class Pipelines
 
     private static IResourceLoader<SMAPIModLoadoutItem.ReadOnly, SMAPIManifest> CreateManifestPipeline(IFileStore fileStore)
     {
-        var pipeline = new FileStoreLoader(fileStore)
+        var pipeline = new FileStoreStreamLoader(fileStore)
             .ThenDo(Unit.Default, static (_, _, resource, _) =>
             {
-                var bytes = resource.Data;
-                var json = Encoding.UTF8.GetString(bytes);
+                using var streamReader = new StreamReader(stream: resource.Data, encoding: Encoding.UTF8);
+                var json = streamReader.ReadToEnd();
 
                 var manifest = Interop.SMAPIJsonHelper.Deserialize<SMAPIManifest>(json);
                 ArgumentNullException.ThrowIfNull(manifest);

